include/arrays/span.hpp

```diff
@@ -19,6 +19,7 @@ class span {
 
 public:
   inline span(T* ptr, size_t size) : ptr_(ptr), size_(size) { }
+  inline span(): span((T*) "", 0) {}
```

Comment:
```suggestion
  inline span(): span(nullptr, 0) {}
```
Is there a reason to use a C-style cast?

Reply:
True, a nullpointer would do as well. The idea was to use a valid static allocation to avoid issues with derferencing a nullpointer, but this will not happen with an empty span anyway.